artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java

```diff
@@ -722,6 +722,23 @@ public void testSetNestedPropertyOnExistingCollectionEntryViaMappedNotation() th
       Assert.assertEquals("TF", configuration.getConnectionRouters().get(0).getKeyFilter());
    }
 
+
+   @Test
+   public void testAddressViaProperties() throws Throwable {
+      ConfigurationImpl configuration = new ConfigurationImpl();
+
+      Properties properties = new Properties();
+
+      properties.put("addressConfigurations.\"LB.TEST\".queueConfigs.\"LB.TEST\".routingType", "ANYCAST");
```

Comment:
If you add another queue property configuration.getAddressConfigurations().get(0).getQueueConfigs().size() will be 2 because QueueConfiguration name is a SimpleString
`properties.put("addressConfigurations.\"LB.TEST\".queueConfigs.\"LB.TEST\".durable", "true");`

Reply:
that is a problem, let me revisit.... not sure if the lookup needs to be smplestring aware of whether name should always be a string or both!
Probably ensuring the name accessor is a string and hiding the simple string as an implementation detail is best

Reply:
the latter is a simpler change too.

Reply:
excellent review, thanks!